src/main/java/core/basesyntax/operation/BalanceHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class BalanceHandler implements OperationHandler {
+    private ShopService shopService;
+
+    public BalanceHandler(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.BALANCE) {
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            shopService.addFruits(fruit, quantity);
+        } else {
+            throw new RuntimeException("Unsupported operation for BalanceHandler");
+        }

```

Comment:
The operation type check is not necessary in this class. This check should be performed before calling the `apply` method, for example in the `OperationStrategyImpl` class. This way, you can be sure that the correct operation is passed to the correct handler, and you can avoid unnecessary checks in each handler.